Ros2 safety area manager

MarlonRiv opened the pull request:
Safety area manager validates if the position of the UAV is valid, and this info is shared directly in the diagnostics msg, which we can use instead of validating the current position in autostart with the validateReference method.

MarlonRiv committed:
Using safety area manager diagnostics for the position validation

MarlonRiv committed:
Merge branch 'ros2' into ros2_safety_area_manager

MarlonRiv committed:
Merge branch 'ros2' into ros2_safety_area_manager

MarlonRiv committed:
Applied new formatting to changes

klaxalk commented on src/automatic_start.cpp:
```diff
@@ -549,7 +551,90 @@ void AutomaticStart::timerMain() {
       }
 
-      return;
+      // | -------------------- ready to takeoff -------------------- |
```
This whole block of code (50+ lines down from here) should not be here, probably a result of a bad merge?

MarlonRiv replied:
Oh yes, most likely a bad merge, fixing now!

klaxalk commented on src/automatic_start.cpp:
```diff
@@ -16,6 +16,7 @@
 #include <mrs_msgs/msg/control_manager_diagnostics.hpp>
+#include <mrs_msgs/msg/safety_area_manager_diagnostics.hpp>
 #include <mrs_msgs/msg/uav_manager_diagnostics.hpp>
 #include <mrs_msgs/srv/validate_reference.hpp>
```
If I understand the changes well, the `validate_reference.hpp` should not be needed anymore.

klaxalk commented on src/automatic_start.cpp:
```diff
@@ -748,36 +833,6 @@ bool AutomaticStart::takeoff() {
-  }
-}
-
 //}
```
pls remove the end fold tag

klaxalk commented on src/automatic_start.cpp:
```diff
@@ -549,7 +551,90 @@ void AutomaticStart::timerMain() {
         return;
       }
 
-      return;
```
the `return` should stay

klaxalk commented on src/automatic_start.cpp:
The `service_client_validate_reference_` should probably be removed, if I understand the changes well.

klaxalk reviewed (commented):
Hey @MarlonRiv, could you please fix these minor things? Also, I recommennd to merge the current `ros2` into it (should not be problematic). Thanks!

MarlonRiv committed:
Comments from review

-Removed unncessary validate reference msg include, service client.
-Fixing typos and dangling code blocks from bad merge

MarlonRiv committed:
Merge branch 'ros2' into ros2_safety_area_manager

Commit:
updated world config in tests

Commit:
Merge remote-tracking branch 'origin/ros2' into ros2_safety_area_manager
